www/caddy/src/opnsense/mvc/app/models/OPNsense/Caddy/Caddy.php

```diff
@@ -107,6 +108,52 @@ private function checkSubdomainsAgainstDomains($subdomains, $domains, $messages)
         }
     }
 
+    // Get the OPNsense WebGUI configuration and store it
+    private $webgui;
+
+    private function getWebGui() {
+        if (!$this->webgui) {
+            $this->webgui = Config::getInstance()->object()->system->webgui ?? null;
+        }
+        return $this->webgui;
+    }
+
+    // Get the default OPNsense WebGUI ports
+    private function getWebGuiPorts() {
+        $webgui = $this->getWebGui();
+        $webGuiPorts = [];
+
+        // Only add ports to array if no specific interfaces for the WebGUI are set
+        if (empty($webgui->interfaces)) {
+            // Add port 443 if no specific port is set, only checks for default WebGUI settings
+            if (empty($webgui->port)) {
+                $webGuiPorts[] = '443';
+            }
+
+            // Add port 80 if HTTP redirect is not explicitly disabled
+            if (!isset($webgui->disablehttpredirect) || $webgui->disablehttpredirect != '1') {
```

Comment:
```
if (!empty((string)$webgui->disablehttpredirect)) {
```

Reply:
I had to go with:

```if (empty((string)$webgui->disablehttpredirect)) {```

Because the string is empty, when the httpredirect is enabled, but it is !empty, when it is disabled (its 1 in the config then)